Concept docs/the remaining ones (#56)
SenseNet/docs.sensenet.com: content/concepts/basics/05-rest-api.md

metaTitle: "sensenet - REST API"
metaDescription: "sensenet REST API"
---

sensenet offers a rich API allowing you to manage, integrate and deliver content on all types of clients, devices and platforms. This section explains the API structure and main concepts in more depth.

sensenet content repository is an [OData](https://www.odata.org/) producer which means your application can consume our [OData](https://www.odata.org/) service to create web and mobile apps or any other type of content based application. It supports CRUD operations, properties, binary streams, paging options, custom queries and many more.

> The requested resource can be any content in the repository (since everything is a content in sensenet, it could be really anything) that is permitted for the current user.

Author: ```suggestion
```

metaTitle: "sensenet - REST API"
metaDescription: "sensenet REST API"
---

> sensenet offers you a rich API allowing you to manage, integrate and deliver content on all types of clients, devices and platforms. This section explains the API structure and main concepts in more depth.

Reviewer: ```suggestion
sensenet offers a rich API allowing you to manage, integrate and deliver content on all types of clients, devices and platforms. This section explains the API structure and main concepts in more depth.
```

> sensenet content repository is an [OData](https://www.odata.org/) producer which means your application can consume our [OData](https://www.odata.org/) service to create web and mobile apps or any other type of content based applications. It supports CRUD operations, properties, binary streams, paging options, custom queries and many more.

Reviewer: ```suggestion
sensenet content repository is an [OData](https://www.odata.org/) producer which means your application can consume our [OData](https://www.odata.org/) service to create web and mobile apps or any other type of content based application. It supports CRUD operations, properties, binary streams, paging options, custom queries and many more.
```

The requested resource can be any content in the repository (since everything is a content in sensenet, it could be really anything) that is permitted for the current user.

[dig deeper into sensenet REST API](/api-docs/basic-concepts)

> ## Content delivery

Reviewer: ```suggestion
# Content delivery
```

> The requested resource can be any content in the repository that is permitted for the current user. Since everything is a content in sensenet this means that every item (user, folder, document, etc) can be accessed through the same API.

Reviewer: it seems like a duplication of line 11

> Resources can be accessed as individual entities or as collections (containers) of content. Bases on the way of accessing the result will contain a content's or its children's metadata. The reponse is always in json format which you can easily process and built into your own application.

Reviewer: ```suggestion
Resources can be accessed as individual entities or as collections (containers) of content. Bases on the way of accessing the result will contain a content's or its children's metadata. The response is always in json format which you can easily process and build into your own application.
```

Reviewer: Bases on the way of accessing the result will contain a content's or its children's metadata.

I don't understand this sentence, could you shed some light on it please.

[more about content delivery](/api-docs/basic-concepts/01-entry)

> ## Content management and beyond

Reviewer: ```suggestion
# Content management and beyond
```

> sensenet's REST API not only a content delivery API, it supports also CRUD operations and many other things for managing content like upload, versioning, approval or setting permissions.

Reviewer: ```suggestion
sensenet's REST API is not only a content delivery API, it supports also CRUD operations and many other things for managing content like upload, versioning, approval or setting permissions.
```

[explore all the possibilities](/api-docs/content-management)

> ## Accessibility

Reviewer: ```suggestion
# Accessibility
```

> All content in the repository is secured by the permission system, which means users can only access or manage those content that is allowed for them. This makes also the API super secure and flexible in the same time, you define the access points and levels and these settings also apply to every request and response sent and accepted through the API.

Reviewer: ```suggestion
All content in the repository is secured by the permission system, which means users can only access or manage content that is allowed for them. This also makes the API super secure and flexible in the same time. You define the access points and levels and these settings also apply to every request and response sent and received through the API.
```

> ## Extend the API

Reviewer: ```suggestion
# Extend the API
```

> sensenet's REST API is also expandable in many ways. Possiblity of filtering, querying or ordering the results provided by sensenet services give you all the tools you need to add custom business logic to your API calls. Additionaly it is possible to add custom APIs (custom [Actions](/concepts/basics/08-actions)) if you have the on premise version.

Reviewer: ```suggestion
sensenet's REST API is also expandable in many ways. Possiblity of filtering, querying or ordering the results provided by sensenet services give you all the tools you need to add custom business logic to your API calls. Additionally it is possible to add custom APIs (custom [Actions](/concepts/basics/08-actions)) if you have the on premise version.
```
